pkg/asset/quota/openstack/openstack.go

```diff
@@ -0,0 +1,119 @@
+package openstack
+
+import (
+	"sort"
+
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	"github.com/openshift/installer/pkg/asset/installconfig/openstack/validation"
+	"github.com/openshift/installer/pkg/quota"
+	"github.com/openshift/installer/pkg/types"
+	openstackprovider "sigs.k8s.io/cluster-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
+)
+
+// Constraints returns a list of quota constraints based on the InstallConfig.
+// These constraints can be used to check if there is enough quota for creating a cluster
+// for the install config.
+func Constraints(config *types.InstallConfig, controlPlanes []machineapi.Machine, computes []machineapi.MachineSet) []quota.Constraint {
+	ctrplConfigs := make([]*openstackprovider.OpenstackProviderSpec, len(controlPlanes))
+	for i, m := range controlPlanes {
+		ctrplConfigs[i] = m.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec)
+	}
+	computeReplicas := make([]int64, len(computes))
+	computeConfigs := make([]*openstackprovider.OpenstackProviderSpec, len(computes))
+	for i, w := range computes {
+		computeReplicas[i] = int64(*w.Spec.Replicas)
+		computeConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*openstackprovider.OpenstackProviderSpec)
+	}
+
+	var ret []quota.Constraint
+	for _, gen := range []constraintGenerator{
+		controlPlane(config, ctrplConfigs),
+		compute(config, computeReplicas, computeConfigs),
+		others,
+	} {
+		ret = append(ret, gen()...)
+	}
```

Comment:
I think we're being too cute here. Change `controlPlane` and `compute` to return a slice of constraints instead of a function.
```suggestion
	ret := controlPlane(config, ctrplConfigs)
	ret = append(ret, compute(config, computeReplicas, computeConfigs)...)
```

Reply:
ack, I'll take that suggestion, thanks.